### ISSUE TITLE
should mutable_tree_map inherit from Mutable_Map instead of Map

### ISSUE THREAD

**Issue by michaellilltokiwa** [tokiwa-software/fuzion#4324]

**Comment by michaellilltokiwa** [comment]
> in my opinion, yes, but also `Mutable_Map` should inherit from `Map`

We can not inherit in private yet. Map is immutable so Mutable_Map should not show to the outside that it inherits from Map, IMHO.

**Comment by fridis** [comment]
Yes. For the relation to `Map`: there should be a feature to create an immutable `Map`. It might even be possible to have mutable data structures inherit from their immutable counterparts, with the caveat that the features, even those that only read data, require the a mutate effect that is not present in the immutable version. So I am not sure if the benefit outweighs the confusion this might cause.
